tests/sync/test_client.py

```diff
@@ -115,3 +114,61 @@ def test_should_discover_ip():
     hostport = client.hostport
 
     assert '0.0.0.0:0' != hostport
+
+
+def test_sync_register_json():
+    client = TChannel('test-client')
+
+    @client.json.register
+    def hello(request):
+        return {}
+
+    assert not client._dep_tchannel._handler.handlers.get('hello', None)
+
+    client.json.register(hello)
+    assert not client._dep_tchannel._handler.handlers.get('hello', None)
```

Comment:
This seems like a janky way to test this. With that number of `_` and `.`, there has to be a better way.

Try creating a parameterized fixture (see http://pytest.org/latest/example/parametrize.html) that returns a function that takes a tchannel and registers endpoints on it. So, parameterize the fixture over whether you want raw/json/thrift endpoints (and perhaps a "direct" one which just uses tchannel.register) and return the appropriate function.

Then in the test, call that function on one sync tchannel and one async tchannel. Ensure that you're able to make requests to the async tchannel and that requests to the sync tchannel fail with unknown method or whatever we have there.